jib-core/src/main/java/com/google/cloud/tools/jib/builder/steps/RetrieveRegistryCredentialsStep.java

```diff
@@ -82,22 +87,29 @@ static RetrieveRegistryCredentialsStep forTargetImage(
   @Override
   @Nullable
   public Credential call() throws CredentialRetrievalException {
-    buildLogger.lifecycle(String.format(DESCRIPTION, registry) + "...");
+    String description = makeDescription(registry);
+    emit(LogEvent.lifecycle(description + "..."));
 
-    try (Timer ignored = new Timer(buildLogger, String.format(DESCRIPTION, registry))) {
+    try (Timer ignored =
+        LoggingTimer.newTimer(logMessage -> emit(LogEvent.debug(logMessage)), description)) {
```

Comment:
IMHO the base code shouldn't decide that timer events should be logged (or lifecycle events, for that matter).  Rather this should be emitting `TimerEvent` messages. The logger should decide whether it wants to consume and report those messages.

Reply:
That is a better design - I'll change to that.